### ISSUE TITLE
E722: not reported when raising original exception

### ISSUE THREAD

**Issue by matejsp** [astral-sh/ruff#17827]
### Summary

When raising the original exception the ruff does not report the use of bare except. However flake8 does.

```
try:
    pass
except:
    raise
```

```
ruff check test.py
All checks passed!
```

```
flake8 test.py
test.py:4:1: E722 do not use bare 'except'
```

https://play.ruff.rs/fb125639-d561-407e-afca-16b2fd1240cd

In our case it was error prone because in our except we were handling errors in except block:

```
except:
   if input == 1:
     return x
   raise
```

### Version

ruff 0.11.6

**Comment by ntBre** [comment]
It looks like this was changed in #1124 to address #1119 with the logic that re-raising the exception solves the problems described in the rule [docs](https://docs.astral.sh/ruff/rules/bare-except/). So I would say your playground example is working as intended, but this still seems like a potential problem when the exception is only _conditionally_ re-raised, as in your example here. This is discussed a  bit [here](https://github.com/PyCQA/flake8-bugbear/issues/174) too.

@dylwil3 is currently working on better control-flow analysis that could help to check conditional branches here.
